Lecture 3 homework

Bepely opened the pull request:
• Added header and footer
• Added Layout
• Added Tabs functionality
• Reworked Restaurants display functionality

Bepely committed:
Lecture 3 homework

• Added header and footer
• Added Layout
• Added Tabs functionality
• Reworked Restaurants display functionality

zubkov7 commented:
прогоните пожалуйста prettier, чтобы отформатировать текст

zubkov7 commented on src/app.jsx:
```diff
@@ -0,0 +1,31 @@
+            <Layout>
+            {/* Display selection tabs for each restaraunt */}
+            {restaurants.map((restaurant) => {
+              return <>
```
здесь лишний фрагмент

zubkov7 commented on src/counter/component.jsx:
```diff
@@ -0,0 +1,28 @@
+import { useState } from "react";
+
+function Counter() {
```
значения минимума и максимума обычно выносят в отдельные константы

zubkov7 commented on src/tab/component.jsx:
```diff
@@ -0,0 +1,15 @@
+function Tab({restaurant, setCurrentRestaurant}) {
```
компонент Tab должен быть максимально абстрактным, поэтому лучше, чтобы он принимал абстрактные пропсы: обработчик onClick (потребитель сам определит логику) и text для отображения

zubkov7 commented on src/app.jsx:
```diff
@@ -0,0 +1,31 @@
+
+function App() {
+
+    const [currentRestaurant, setCurrentRestaurant] = useState(restaurants[0])
```
сейчас при повторном клике на таб, вызовется функция setCurrentRestaurant. Нужно обработать этот кейс и не вызывать ее. Можно добавить условие в функцию, можно повесить disabled на активную табу

Bepely committed:
Lecture 3 homework 2

• Added tabs disable functionality
• Added constants Min and Max in counter
• Removed redundant fragment

Bepely committed:
Lecture 3 homework #3

• Updated button disabled functionality

zubkov7 commented:
прогоните пожалуйста prettier, чтобы отформатировать код

zubkov7 commented on src/app.jsx:
```diff
@@ -0,0 +1,32 @@
+              return  <Tab 
+              callback={()=>{setCurrentRestaurant(restaurant)}} 
+              text={restaurant.name} 
+              disabled={currentRestaurant.name == restaurant.name ? true : false}/>
```
лучше применять строгое сравнение ===, чтобы еще проверялся тип

также можно просто currentRestaurant.name === restaurant.name, это и так вернет нужное значение boolean

zubkov7 commented on src/counter/component.jsx:
```diff
@@ -0,0 +1,31 @@
+    // Create a counter for added dishes
+    const [itemsCounter, setItemsCounter] = useState(0);
+
+    const itemsMin = 0;
```
лучше неизменяемые константы выносить из компонента, чтобы они не создавались на каждый рендер
